drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -412,63 +420,67 @@ local function device_type_supports_button_switch_combination(device, endpoint_i
   return false
 end
 
-local function get_first_non_zero_endpoint(endpoints)
-  table.sort(endpoints)
-  for _,ep in ipairs(endpoints) do
-    if ep ~= 0 then -- 0 is the matter RootNode endpoint
-      return ep
+--- @param cluster_id integer|nil cluster id to check for (find first non-root ep if nil)
+--- @param opts table|nil currently, only valid option is feature_bitmap to specify cluster feature support
+--- @return integer default_endpoint_id the first cluster-specified ep, or the first non-root ep, or the root ep
+local function find_default_endpoint(device, cluster_id, opts)
+  local get_first_cluster_endpoint = function(endpoints_list)
+    table.sort(endpoints_list)
+    for _, ep_id in ipairs(endpoints_list) do
+      if ep_id ~= 0 then return ep_id end -- ignore Matter RootNode when choosing default endpoint
     end
+    return -1
   end
-  return nil
-end
-
---- find_default_endpoint is a helper function to handle situations where
---- device does not have endpoint ids in sequential order from 1
-local function find_default_endpoint(device)
-  if device.manufacturer_info.vendor_id == AQARA_MANUFACTURER_ID and
-     device.manufacturer_info.product_id == AQARA_CLIMATE_SENSOR_W100_ID then
-    -- In case of Aqara Climate Sensor W100, in order to sequentially set the button name to button 1, 2, 3
-    return device.MATTER_DEFAULT_ENDPOINT
+  local get_default_endpoint = function()
+    local all_endpoint_ids = device:get_endpoints()
+    table.sort(all_endpoint_ids)
+    for _, ep_id in ipairs(all_endpoint_ids) do
+      if ep_id ~= 0 then return ep_id end -- ignore Matter RootNode when choosing default endpoint
+    end
+    return 0 -- return root node if all else fails
   end
 
+  -- Return first switch ep if both switch and button eps are present and it is a profile-supported device type
   local switch_eps = device:get_endpoints(clusters.OnOff.ID)
-  local button_eps = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.MOMENTARY_SWITCH})
-
-  -- Return the first switch endpoint as the default endpoint if no button endpoints are present
-  if #button_eps == 0 and #switch_eps > 0 then
-    return get_first_non_zero_endpoint(switch_eps)
+  if #switch_eps then
```

Comment:
```suggestion
  if #switch_eps > 0 then
```
You might need `> 0` since 0 would evaluate to true

Reply:
true, thanks!